### PR TITLE
Add `push` and `pull` steps for Azure Blob Storage

### LINKED ISSUE #82: Create Azure Blob Storage Push Step
Add a project step that allows users to push their project to an Azure Blob Storage container at a given path. Should accept credentials that can be used for authentication with Azure.

### LINKED ISSUE #83: Create Azure Storage Pull Step
Add a project step that allows users to pull their project from a Azure Blob Storage container at a given path. Should accept credentials that can be used for authentication with Azure.

### PR DESCRIPTION
**Pull request by desertaxle**
- Adds `push_to_azure_blob_storage` and `pull_from_azure_blob_storage` project steps to allow Azure Blob Storage to be used for deployment storage and retrieval. New steps follow the pattern established for existing `push` and `pull` steps.
- Cleans up docstrings and field description for `AzureBlobStorageCredentials`
- Adds logo and documentation URLs to `AzureContainerWorker`

Closes #82 
Closes #83

### Example

```yaml
prefect_version: ...
name: ...

push:
    - prefect_azure.deployments.steps.push_to_azure_blob_storage:
        requires: prefect-azure
        container: my-container
        folder: my-project
        credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"

pull:
    - prefect_azure.deployments.steps.pull_from_azure_blob_storage:
        requires: prefect-azure
        container: "{{ container }}"
        folder: "{{ folder }}"
        credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"
```

### Screenshots

localhost_8000_prefect-azure_deployments_steps_

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-azure/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-azure/blob/main/CHANGELOG.md)

### CODE REVIEW

File: tests/projects/test_steps.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    return tmp_path
+
+
+class TestPushProject:
+    @pytest.mark.usefixtures("mock_azure_blob_storage")
+    def test_push_project_to_azure_blob_storage_with_connection_string(
+        self, tmp_files: Path, container_client_mock: MagicMock
+    ):
+        container = "test-container"
+        folder = "test-folder"
```
Comment (serinamarie) [inline tests/projects/test_steps.py:70]: are trailing slashes handled?

**Review by serinamarie — COMMENTED** [review]
Nice! 

For the tests I was wondering regarding folders:
1. Are trailing slashes in the `folder` arg handled?
2. Should we test that subfolders are properly read from/to?
3. It seems like all of these tests include a `folder` arg. Should we test not including a folder?

**Comment by desertaxle** [comment]
@serinamarie I added some test coverage based on your suggestions and removed references to 'projects' in this PR, so it should be ready for another review.

File: prefect_azure/deployments/steps.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    pull:
+        - prefect_azure.deployments.steps.pull_from_azure_blob_storage:
+            requires: prefect-azure[blob_storage]
+            container: "{{ container }}"
+            folder: "{{ folder }}"
+            credentials: "{{ prefect.blocks.azure-blob-storage-credentials.dev-credentials }}"
+    ```
+
+For more information about using deployment steps, check out out the Prefect [docs](https://docs.prefect.io/latest/concepts/projects/#the-prefect-yaml-file).
```
Comment (jakekaplan) [inline prefect_azure/deployments/steps.py:31]: Is this the right link?

File: prefect_azure/deployments/steps.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+    ```yaml
+    prefect_version: ...
+    name: ...
+
+    push:
+        - prefect_azure.deployments.steps.push_to_azure_blob_storage:
+            requires: prefect-azure[blob_storage]
+            container: my-container
+            folder: my-project
```
Comment (jakekaplan) [inline prefect_azure/deployments/steps.py:20]: not sure if you want to call it something different than `my-project` to differentiate from "projects" at all 🤷‍♂️

**Pull request merged**